ProjFS.Mac/PrjFSKext/PrjFSKext/KauthHandler.cpp

```diff
@@ -277,6 +277,30 @@ void KauthHandler_AbortOutstandingEventsForProvider(VirtualizationRootHandle pro
     Mutex_Release(s_outstandingMessagesMutex);
 }
 
+static void UseMainForkIfNamedStream(
+    // In+out params:
+    vnode_t& vnode,
+    // Out params:
+    bool&    putVnodeWhenDone)
+{
+    // A lot of our file checks such as attribute tests behave oddly if the vnode
+    // refers to a named fork/stream; apply the logic as if the vnode operation was
+    // occurring on the file itself. (/path/to/file/..namedfork/rsrc)
+    if (vnode_isnamedstream(vnode))
+    {
+        vnode_t mainFileFork = vnode_getparent(vnode);
+        assert(NULLVP != mainFileFork);
+        
+        if (putVnodeWhenDone)
+        {
+            vnode_put(vnode);
+        }
```

Comment:
What's the purpose of having this block of code?

```
        if (putVnodeWhenDone)
        {
            vnode_put(vnode);
        }
```

The callers of `UseMainForkIfNamedStream` are taking care of calling `vnode_put`

Reply:
The intention is that since we're replacing the existing (incoming) vnode under consideration, we should also release it if needed. This would in principle be needed on a code path using `vnode_lookup()`. I subsequently decided that we probably won't see any named forks on our only code path where we use that though. So yeah, it's a toss-up over whether to leave the functionality in the helper function in case we need it in future or take it back out for now.

Reply:
I've removed this functionality for now as it's not being used.

Reply:
Thanks for the details here, all that makes sense.